File: src/main/GO/leetcode/lists/topinterview/LeetCode66.go
package topinterview

func plusOne(digits []int) []int {

	if len(digits) == 0 {
		return []int{}
	}

	flag := 0
	for i := len(digits) - 1; i >= 0; i-- {
		temp := digits[i] + flag
		if i == len(digits)-1 {
			temp += 1
		}
		digits[i] = temp % 10
		flag = temp / 10
	}
	if flag == 1 {
		list := []int{1}
		list = append(list, digits...)
		return list
	}
	return digits
}

File: java/src/leetcode/problems/LeetCode66.java
package leetcode.problems;

/**
 * @author wujingxinit@outlook.com
 * @date 2024/4/11 11:03
 */
public class LeetCode66 {

    public int[] plusOne(int[] digits) {
        int n = digits.length;
        int flag = 0;
        digits[n - 1]++;
        for (int i = n - 1; i >= 0; i--) {
            int sum = digits[i] + flag;
            flag = sum / 10;
            digits[i] = sum % 10;
        }
        if (flag == 1) {
            int[] res = new int[n + 1];
            res[0] = 1;
            for (int i = 1; i <= n; i++) {
                res[i] = digits[i - 1];
            }
            return res;
        }
        return digits;
    }
}
